internal/events/operation_update.go

```diff
@@ -42,6 +42,7 @@ func (em *eventManager) operationUpdateCtx(ctx context.Context, operationID *fft
 	// Special handling for OpTypeTokenTransfer, which writes an event when it fails
 	if op.Type == fftypes.OpTypeTokenTransfer && txState == fftypes.OpStatusFailed {
 		event := fftypes.NewEvent(fftypes.EventTypeTransferOpFailed, op.Namespace, op.ID)
+		// TODO: Need to determine transfer type and track metric
```

Comment:
I think this is now available in `op.Input.getString("type")`

Reply:
Or you can use the helpers in `token_inputs.go` to extract all transfer details from the Input. Maybe that's better...